include/nbl/video/ILogicalDevice.h

```diff
@@ -96,8 +96,13 @@ class ILogicalDevice : public core::IReferenceCounted
     IGPUQueue* getQueue(uint32_t _familyIx, uint32_t _ix)
     {
         const uint32_t offset = (*m_offsets)[_familyIx];
+        return (*m_queues)[offset+_ix].getUnderlyingQueue().get();
+    }
 
-        return (*m_queues)[offset+_ix].get();
+    CThreadSafeGPUQueueAdapter getThreadSafeQueue(uint32_t _familyIx, uint32_t _ix)
```

Comment:
write a comment that using the same queue as both a threadsafe queue and normal queue invalidates the safety.

Reply:
also return a pointer

Reply:
resolved